### ISSUE TITLE
[UseSelect] reach onOuterClick functionality to give right feedback to Formik

### ISSUE THREAD

**Issue by keremciu** [downshift-js/downshift#1191]
- `downshift` version: 5.4.2
- `node` version: 14.15
- `npm` (or `yarn`) version: 6.14.8

**Relevant code or config**

```javascript
```

**What you did**:

Implemented `onBlur` prop to useSelect for Formik field validation.

**What happened**:

If I use `onBlur` provided by useSelect, basically first click to ToggleButton counts as `onBlur` function call and 

**Reproduction repository**:

**Problem description**:

First click to open Menu runs `onBlur` function and naturally value is null then Formik shows an error for this required field.
<img width="1096" alt="Screenshot 2020-10-29 at 16 54 45" src="https://user-images.githubusercontent.com/1593488/97598659-7b42ca00-1a07-11eb-9f98-6a3c686ccb6c.png">

Basically first problem is `onBlur` running for ToggleButton click and second problem tis there's no `onBlur` function call for `onOuterClick` case that's why I'm not able to run validation on the formik field.

**Suggested solution**:

https://github.com/downshift-js/downshift/blob/master/src/hooks/useSelect/index.js#L181

This place looks right to put `onOuterClick` prop. Users can benefit from it like `<Downshift>` component.

or It would be nice to know how to avoid first Button click from provided `onBlur` function.

**Comment by keremciu** [comment]
I implemented a prop called `onMenuBlur` to get over this problem but somehow it's triggered even I click on a menu item. I'm trying to find how to overcome that part, looks like now it's Formik implementation detail.
<img width="186" alt="Screenshot 2020-10-29 at 18 59 09" src="https://user-images.githubusercontent.com/1593488/97613323-d4672980-1a18-11eb-9e91-65ea878b4786.png">

**Comment by keremciu** [comment]
<img width="476" alt="Screenshot 2020-10-29 at 20 18 00" src="https://user-images.githubusercontent.com/1593488/97622104-147fd980-1a24-11eb-9026-d39385e983bc.png">

I trigger validation two times to reach right evaluation. onMenuBlur always give me same event.target, I couldn't find a way to say `don't run setTouched if user clicks on menu item`

**Comment by silviuaavram** [comment]
Maybe you don’t need any of this. What are you trying to achieve? When do
you want to run your validation? If you want on menu blur, why not use
onStateChange and capture MenuBlur state change type?

**Comment by silviuaavram** [comment]
https://github.com/downshift-js/downshift/tree/master/src/hooks/useSelect#onstatechange

Because you don’t probably want to handle the event. You just want to react
to a state change, which is useSelect.stateChangeTypes.MenuBlur.
